lib/pbench/test/unit/server/query_apis/test_namespace_and_rows.py

```diff
@@ -764,20 +765,33 @@ def test_get_index(self, attach_dataset, provide_metadata):
         indices = self.cls_obj.get_index(drb, self.index_from_metadata)
         assert indices == "unit-test.v5.result-data-sample.2020-08"
 
-    def test_exceptions_on_get_index(self, attach_dataset):
+    def test_get_index_none(self, attach_dataset):
         test = Dataset.query(name="test")
 
-        # When server index_map is None we expect 500
-        with pytest.raises(APIAbort) as exc:
-            self.cls_obj.get_index(test, self.index_from_metadata)
-        assert exc.value.http_status == HTTPStatus.INTERNAL_SERVER_ERROR
+        # No index map at all for dataset
+        assert self.cls_obj.get_index(test, self.index_from_metadata) == ""
 
-        Metadata.setvalue(
+        # An index map that doesn't contain the desired root
+        IndexMap.create(
             dataset=test,
-            key=Metadata.INDEX_MAP,
-            value={"unit-test.v6.run-data.2020-08": ["random_md5_string1"]},
+            map={
+                "result-data": {
+                    "unit-test.v6.result-data.2020-09": ["random_md5_string"]
+                }
+            },
         )
-
-        # When server index_map doesn't have mappings for result-data-sample
-        # documents we expect the indices to an empty string
         assert self.cls_obj.get_index(test, self.index_from_metadata) == ""
+
+        # An index map that contains the desired root plus others
+        IndexMap.create(
+            dataset=test,
+            map={
+                "result-data-sample": {
+                    "unit-test.v6.result-data-sample.2020-08": ["random_md5_string1"]
+                }
+            },
+        )
+        assert (
+            self.cls_obj.get_index(test, self.index_from_metadata)
+            == "unit-test.v6.result-data-sample.2020-08"
+        )
```

Comment:
The "desired root" is `index_from_metadata`, right?  But, what line 786 adds is `"result-data-sample"`...it might allay some confusion if the code referred to it in the same way in both places.

The "plus others" comes from the `IndexMap.create()` operation at line 775, right?  (It would be nice if the comment made that clearer.)

Reply:
Your comment on the comment points out the fact that the comment *should* probably have been correct, but got broken. That is, there aren't any other indices or roots in the map.

This also kinda points out that the original test case includes multiple cases only one of which has any relationship to the test method *name*, and the new name doesn't actually help any ... 😦